btcutil/psbt/taproot.go

```diff
@@ -93,6 +94,43 @@ func (s *TaprootBip32Derivation) SortBefore(other *TaprootBip32Derivation) bool
 	return bytes.Compare(s.XOnlyPubKey, other.XOnlyPubKey) < 0
 }
 
+// minTaprootBip32DerivationByteSize returns the minimum number of bytes
+// required to encode a Taproot BIP32 derivation field, given the number of
+// leaf hashes.
+//
+// NOTE: This function does not account for the size of the BIP32 child indexes,
+// as we are only computing the minimum size (which occurs when the path is
+// empty). The bits package is used to safely detect and handle overflows.
+func minTaprootBip32DerivationByteSize(numHashes uint64) (uint64, error) {
+	// The Taproot BIP32 derivation field is encoded as:
+	//   [1 byte: number of leaf hashes]
+	//   [N × 32 bytes: leaf hashes]
+	//   [4 bytes: master key fingerprint]
+	//   [1 byte: number of BIP32 child indexes]
+	//   [M × 4 bytes: BIP32 child indexes]
+	//
+	// To compute the minimum size, we assume:
+	// - N = numHashes (provided)
+	// - M = 0 (no child indexes)
+	//
+	// So the base size is:
+	//   1 (leaf hash count) + (N × 32) + 4 (fingerprint) + 1 (empty path count)
+	//
+	// First, we calculate the total number of bytes for the leaf hashes.
+	totalHashesBytes, mulCarry := bits.Mul64(numHashes, 32)
+	if mulCarry != 0 {
+		return 0, ErrInvalidPsbtFormat
+	}
+
+	// Add 6 bytes: 1 (leaf hash count) + 4 (fingerprint) + 1 (path length)
+	result, addCarry := bits.Add64(totalHashesBytes, 6, 0)
```

Comment:
The path length is defined as `<32-bit little endian uint path element>*`, meaning zero paths is acceptable. That's why `6` is wrong here.